### ISSUE TITLE
Limit patient_id to be a subject_id and not a uuid

### ISSUE THREAD

**Issue by kennsippell** [medic/cht-core#6154]
**What feature do you want to improve?**
The lineage shared library has two inputs for patients: a `patient_id` and a `patient_uuid`. The `patient_id` can be a subject id or a contact's id. For example, [here](https://github.com/medic/cht-core/blob/master/shared-libs/lineage/test/integration.js#L517) in the integration tests where the patient_id is a uuid. 

The downside of this is that when given a patient_id, we need to check if it is a uuid and only if that query fails do we check if it is a subject id. So we are doing two queries to look up this subject id value - this is slow and clutters the already complex hydration code.

**Describe the improvement you'd like**
We should use patient_uuid for uuids and patient_id explicitly subject ids.  Then we can cleanup the hydration code to avoid this unnecessary query. 

**Notes**
I have no idea if this feature is even used or how many forms would need to be migrated.

**Comment by dianabarsan** [comment]
We also do this dance when generating scheduled messages for the admin Message Queue and when displaying the LHS Report list.

I'm afraid that even imposing this rule will not completely clean up the hydration code. 
None of our SMS workflows add `patient_uuid`, so we will still need to query `contacts_by_reference`. 
Also, when `patient_uuid` exists, should we trust it and not check `patient_id` at all ?

**Comment by kennsippell** [comment]
Discussing this for 6.0, I'm worried about how to actually deploy it.  If all your reports have patient_id with uuid, you'd have to:

1. Update the form to use patient_uuid instead
2. Deploy the form
3. Wait for complete rollout
4. Upgrade to 6.x
5. 6.x would run a migration on all the legacy reports? Or how would the old reports still remain linked?

If we want to deploy this, we should consider phasing it somehow and bringing awareness early so forms can be updated before the 6.x release. Deprecation warnings? Wondering if the benefit is worth the upgrade cost. 

Might consider an implementation path like:

1. Add a new field (`patient_id2` but with a good name) which exclusively supports `patient_id` (non-breaking change)
2. Deprecate `patient_id`, such that any use of patient_id results in cht-conf warning. Direct users to use `patient_id2` or `patient_uuid`
3. Give users lots of time (year+)
4. Remove `patient_id` in 6.x

**Comment by dianabarsan** [comment]
I agree with the alternative code path. `patient_id` was always so ambiguous as a field name. 

Overall, I don't think it's a necessarily _bad_ idea to have a fuzzy patient reference field, that can accept multiple values (in case of some weird workflow). 

Unfortunately, renaming `patient_id` to `patient_ref` (`patient_code`?) has implications across all docs, because it would be a bad change if the field on the contact doc is `patient_id` and the field on the report is `patient_ref`, so all contact docs would need updates as well. 

I'm quite keen on renaming 6.0 into a migration bucket.
